src/main/java/org/spongepowered/api/asset/Asset.java

```diff
@@ -77,6 +78,24 @@ default void copyToFile(Path output) throws IOException {
     }
 
     /**
+     * Copes this Asset to the specified 'outputDirectory' {@Link Path}.
+     * @param outputDirectory directory to copy to
+     * @throws IOException
+     */
+    default void copyToDirectory(Path outputDirectory) throws IOException {
+        checkNotNull(outputDirectory, outputDirectory);
+        copyToFile(outputDirectory.resolve(this.getFileName()));
+    }
+
+    /**
+     *
+     * @return get's the last portion of the Asset URL, e.g. the file name
+     */
+    default String getFileName(){
+        return new File(getUrl().getPath()).getName();
```

Comment:
I don't really see a need for constructing a `File` object with the path to an URL here, just split it yourself